mailcom/test/test_utils.py

```diff
@@ -0,0 +1,517 @@
+import math
+from mailcom import utils
+import pytest
+from string import punctuation
+
+
+# these worked when we were using strings
+# with the update to Path, we need to change the tests
+def test_check_dir(tmpdir):
+    mydir = tmpdir.mkdir("sub")
+    assert utils.check_dir(str(mydir))
+    with pytest.raises(OSError):
+        utils.check_dir(str(tmpdir.join("sub2")))
+
+
+def test_make_dir(tmpdir):
+    mydir = tmpdir.join("sub")
+    utils.make_dir(str(mydir))
+    assert mydir.check()
+
+
+def test_check_dir_fail():
+    with pytest.raises(OSError):
+        utils.check_dir(str("mydir"))
+
+
+# test cases for email language detection
+langid_langs = [
+    "af",
+    "am",
+    "an",
+    "ar",
+    "as",
+    "az",
+    "be",
+    "bg",
+    "bn",
+    "br",
+    "bs",
+    "ca",
+    "cs",
+    "cy",
+    "da",
+    "de",
+    "dz",
+    "el",
+    "en",
+    "eo",
+    "es",
+    "et",
+    "eu",
+    "fa",
+    "fi",
+    "fo",
+    "fr",
+    "ga",
+    "gl",
+    "gu",
+    "he",
+    "hi",
+    "hr",
+    "ht",
+    "hu",
+    "hy",
+    "id",
+    "is",
+    "it",
+    "ja",
+    "jv",
+    "ka",
+    "kk",
+    "km",
+    "kn",
+    "ko",
+    "ku",
+    "ky",
+    "la",
+    "lb",
+    "lo",
+    "lt",
+    "lv",
+    "mg",
+    "mk",
+    "ml",
+    "mn",
+    "mr",
+    "ms",
+    "mt",
+    "nb",
+    "ne",
+    "nl",
+    "nn",
+    "no",
+    "oc",
+    "or",
+    "pa",
+    "pl",
+    "ps",
+    "pt",
+    "qu",
+    "ro",
+    "ru",
+    "rw",
+    "se",
+    "si",
+    "sk",
+    "sl",
+    "sq",
+    "sr",
+    "sv",
+    "sw",
+    "ta",
+    "te",
+    "th",
+    "tl",
+    "tr",
+    "ug",
+    "uk",
+    "ur",
+    "vi",
+    "vo",
+    "wa",
+    "xh",
+    "zh",
+    "zu",
+]
+lang_samples = {
+    "J'espère que tu vas bien! "
+    "Je voulais partager avec toi quelques photos de mon dernier voyage!": "fr",
+    "Hola, ¿cómo estás? Espero que estés bien. ¡Hasta pronto!": "es",
+    "Hello, how are you? I hope you are well. See you soon!": "en",
+    "Hallo, wie geht es dir? Ich hoffe, es geht dir gut. Bis bald!": "de",
+    "Ciao, come stai? Spero che tu stia bene. A presto!": "it",
+    "Olá, como você está? Espero que você esteja bem. Até logo!": "pt",
+    "Привет, как дела? Надеюсь, у тебя все хорошо. Увидимся скоро!": "ru",
+    "你好，你好吗？希望你一切都好。很快见到你！": "zh",
+    "こんにちは、元気ですか？ あなたが元気であることを願っています。 またね！": "ja",
+    "안녕하세요, 어떻게 지내세요? 잘 지내길 바랍니다. 곧 뵙겠습니다!": "ko",
+    "مرحبًا، كيف حالك؟ آمل أن تكون بخير. أراك قريبًا!": "ar",
+    "नमस्ते, कैसे हो? आशा है कि आप ठीक होंगे। जल्द ही मिलेंगे!": "hi",
+}
+punctuations_as_str = "".join(punctuation)
+single_detect_threshold = 0.7
```

Comment:
I would write the constants name in this line and the two below with all caps to make clear that these are constants and not variables.